src/go/rpk/pkg/cli/cluster/config/set.go

```diff
@@ -60,92 +78,152 @@ You may also use <key>=<value> notation for setting configuration properties:
-				if value.Restart {
-					fmt.Print("\nCluster needs to be restarted. See more details with 'rpk cluster config status'.\n")
+		if !ok {
+			// loop over schema, try to find key in the Aliases,
+			for _, v := range schema {
+				if slices.Contains(v.Aliases, key) {
+					meta, ok = v, true
 					break
 				}
 			}
+			if !ok {
+				return nil, nil, fmt.Errorf("unknown property %q", key)
+			}
+		}
+
+		// - For scalars, pass string values through to the REST
+		// API -- it will give more informative errors than we can
+		// about validation.  Special case strings for nullable
+		// properties ('null') and for resetting to default ('')
+		// - For arrays, make an effort: otherwise the REST API
+		// may interpret a scalar string as a list of length 1
+		// (via one_or_many_property).
+
+		if meta.Nullable && value == "null" {
+			// Nullable types may be explicitly set to null
+			upsert[key] = nil
+		} else if meta.Type != "string" && (value == "") {
+			// Non-string types that receive an empty string
+			// are reset to default
+			remove = append(remove, key)
+		} else if meta.Type == "array" {
+			var a anySlice
+			err := yaml.Unmarshal([]byte(value), &a)
+			if err != nil {
+				return nil, nil, fmt.Errorf("invalid list syntax: %v", err)
+			}
+			upsert[key] = a
+		} else {
+			upsert[key] = value
+		}
+	}
+
+	return upsert, remove, nil
+}
+
+func setCloudConfig(ctx context.Context, cfg *config.Config, p *config.RpkProfile, configs []string) (*controlplanev1.UpdateClusterOperation, error) {
+	cloudClient := publicapi.NewCloudClientSet(cfg.DevOverrides().PublicAPIURL, p.CurrentAuth().AuthToken)
+
+	redpandaConfigs := make(map[string]interface{})
+	paths := make([]string, 0)
+
+	for _, cfg := range configs {
+		split := strings.SplitN(cfg, "=", 2)
+		key, value := split[0], split[1]
+		redpandaConfigs[key] = value
+		paths = append(paths, fmt.Sprintf("cluster_configuration.custom_properties.%s", key))
+	}
+	customerProperties, err := structpb.NewStruct(redpandaConfigs)
+	if err != nil {
+		return nil, fmt.Errorf("internal error while converting config to redpanda cloud configs: %v", err)
+	}
+	req := &controlplanev1.UpdateClusterRequest{
+		Cluster: &controlplanev1.ClusterUpdate{
+			Id: p.CloudCluster.ClusterID,
+			ClusterConfiguration: &controlplanev1.ClusterUpdate_ClusterConfiguration{
+				CustomProperties: customerProperties,
+			},
 		},
+		UpdateMask: &fieldmaskpb.FieldMask{Paths: paths},
 	}
 
-	return cmd
+	operation, err := cloudClient.Cluster.UpdateCluster(ctx, connect.NewRequest(req))
+	if err != nil {
+		var ce *connect.Error
+		if errors.As(err, &ce) {
+			if ce.Code() == connect.CodePermissionDenied {
+				return nil, fmt.Errorf("this user does not have permission to update the cluster configuration, please check your role or contact admin")
+			}
+			if ce.Code() == connect.CodeNotFound {
+				return nil, fmt.Errorf("cluster not found. Please ensure the cluster exists in the cloud")
+			}
```

Comment:
This should also handle `400/INVALID_ARGUMENT`.
Sample error:
```json
{
  "code": "INVALID_ARGUMENT",
  "message": "rpc error: code = InvalidArgument desc = some of the customerManagedClusterConfig properties are not allowed",
  "details": [
    {
      "@type": "google.rpc.ErrorInfo",
      "reason": "REASON_INVALID_INPUT",
      "domain": "redpanda.com/controlplane",
      "metadata": {}
    },
    {
      "@type": "google.rpc.BadRequest",
      "field_violations": [
        {
          "field": "cluster_configuration.custom_properties",
          "description": "The properties [invalid_property_asdfasfd] are not allowed"
        }
      ]
    }
  ]
}
```

Reply:
I added the case